src/Utilities/RemediationPairs.js

```diff
@@ -0,0 +1,131 @@
+import chunk from 'lodash/chunk';
+
+const REQUEST_CHUNK_SIZE = 1000;
+const BATCH_REQUEST_SIZE = 5;
+const REQUEST_INTERVAL = 15000; //15 econds. AKAMAI allowed life for 5 API calls
+
+const fetchDataCallback  = (endpoint) => (input) =>
+    fetch(`/api/patch/v1/views${endpoint}`, {
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(input)
+    }).then(result =>
+        result.status === 200
+            ? result.json()
+            : Promise.reject(result.statusText)
+    );
+
+export const transformPairs = (input) => {
+    return {
+        issues: Object.keys(input?.data || {}).map(advisory => {
+            return {
+                id: `patch-advisory:${advisory}`,
+                description: advisory,
+                systems: input.data[advisory]
+            };
+        }
+        )
+    };
+};
+
+const transformSystemsPairs = (input) => {
+    const pairs = [];
+    Object.entries(input.data).map(
+        ([systemID, advisories]) => {
+            advisories.map(advisory => {
+                const pairID = `patch-advisory:${advisory}`;
+                const index = pairs.findIndex(pair => pair.id === pairID);
+                if (index >= 0) {
+                    pairs[index].systems.push(systemID);
+                } else {
+                    pairs.push(
+                        {
+                            id: pairID,
+                            description: advisory,
+                            systems: [systemID]
+                        }
+                    );
+                }
+            });
+        });
+
+    return { issues: pairs };
+};
+
+const batchRequest = async (
+    payload,
+    { responseTransformer, dataFetcher, payloadModifier }
+) => {
+
+    const payloadChunks = chunk(payload, REQUEST_CHUNK_SIZE);
+
+    const paginatedRequest = async (batchIndex, groupChunks) => {
+        const requests = [];
+        for (let i = 0; i < groupChunks.length; i++) {
+            requests.push(
+                await dataFetcher({
+                    ...payloadModifier(payload),
+                    limit: REQUEST_CHUNK_SIZE,
+                    offset: ((batchIndex * BATCH_REQUEST_SIZE) + i) * REQUEST_CHUNK_SIZE
+                }
+                ).then(result => responseTransformer(result))
+            );
+        }
+
+        return await Promise.all(requests);;
```

Comment:
I don't think, you need the `await` or the `Promise.all` here. The Promises pushed into the `request` array are already `await`ed for when looping over the groupChunks and pushed into the array.

Reply:
I thought this will ensure all promises to finish their call. I will verify if they are indeed not necessary and update my PR if needed.

Reply:
@bastilian one more benefit that was in my mind was if I wrap all those API calls promises into a single Promise.all, I would ensure that no remaining promises resolve if any one fails. Considering this, I would keep this. (Feel free to suggest a better way if you have)

Reply:
I'm not 100% certain that is how it would behave. If any of the promises would fail/error and can't be awaited for on line 70, wouldn't they raise and error up there before even reaching the `Promise.all`?

Reply:
hmm, yes it makes sense. I will remove this

Reply:
Done.